modules/gapi/samples/pipeline_modeling_tool.cpp

```diff
@@ -371,7 +380,10 @@ int main(int argc, char* argv[]) {
                     builder.addDummy(call_params, read<DummyParams>(node_fn));
                 } else if (node_type == "Infer") {
                     auto infer_params = read<InferParams>(node_fn);
-                    infer_params.config = config;
+                    RETHROW_WITH_MSG_IF_FAILED(
```

Comment:
Would be great to have a mechanism that could "expand" nested exception with context on current lvl.
@dmatveev, @smirnov-alexey do you have ideas?

e.g: https://en.cppreference.com/w/cpp/error/nested_exception

Reply:
Probably this can be solved by using logs

Reply:
Great idea! Not sure if we need all the stack, but we could try to utilize rethrowing it with a simple `throw;` and catching it on the upper levels (probably with some logs when handling the exception where it is relevant)

Reply:
```
try {
    expr;
} catch (std::exception& e) {
    GAPI_LOG_INFO(MSG); <----- message relevant to current lvl. 
    throw;
}
```